Library/Homebrew/os/mac/version.rb

```diff
@@ -17,16 +17,16 @@ class Version < ::Version
       # and also update references in docs/Installation.md,
       # https://github.com/Homebrew/install/blob/HEAD/install.sh and
       # MacOSVersions::SYMBOLS
-      NEWEST_UNSUPPORTED = "14"
+      NEWEST_UNSUPPORTED = ENV.fetch("HOMEBREW_MACOS_NEWEST_UNSUPPORTED").freeze
       private_constant :NEWEST_UNSUPPORTED
 
       # TODO: bump version when new macOS is released and also update
       # references in docs/Installation.md and
       # https://github.com/Homebrew/install/blob/HEAD/install.sh
-      OLDEST_SUPPORTED = "11"
+      OLDEST_SUPPORTED = ENV.fetch("HOMEBREW_MACOS_OLDEST_SUPPORTED").freeze
       private_constant :OLDEST_SUPPORTED
 
-      OLDEST_ALLOWED = "10.11"
+      OLDEST_ALLOWED = ENV.fetch("HOMEBREW_MACOS_OLDEST_ALLOWED").freeze
```

Comment:
Think these are probably worth moving to `global.rb` after that so we have all these required `ENV.fetch` in a single place.

Reply:
Yeah I initially thought these were macOS only but apparently there's a code path where Linux gets here. global would make it obvious to set the envs on all OS.